src/models/Activity.js

```diff
@@ -22,16 +23,22 @@ const activitySchema = new mongoose.Schema(
           "holiday",
           "cultural",
         ],
-        message: "Invalid category type!",
+        message:
+          "Invalid category type! Choose from ['relaxing', 'creative', 'social', 'educational', 'holiday', 'cultural']",
       },
       required: [true, "please provide a valid category!"],
     },
-    votes: {
-      type: Number,
-      default: 0,
-    },
+    votes: { type: mongoose.Schema.Types.ObjectId, ref: "Vote" },
   },
   { timestamps: true }
 );
 
+// Middleware to create a vote set to 0 before an activity is saved
+activitySchema.pre("save", async function (next) {
+  const newVote = new Vote({ activity: this._id });
+  await newVote.save();
+  this.votes = newVote._id;
+  next();
+});
```

Comment:
If we adjust the `voteSchema` as mentioned below, I think we could remove this as well.

Reply:
This middleware is only responsible for adding a new vote resource associated with an activity resource (whenever a new one is created). So even if once we add the User model, I think it would be a good idea to keep this middleware so that no activity resource is ever without a vote resource.

Reply:
If each `vote` is an association between a user and an activity, then we don't need to create anything. No rows/documents in the collection for that activity just means there are no votes for it. Maybe we can talk more about this in our meeting tomorrow. I don't know that I've done a good job explaining here.

Reply:
This actually is now making more sense to me. I wasn't really thinking of it in the sense that each `vote` is basically an association between a user and activity. Therefore, there needs to be a `POST` and `DELETE` request in order to increase or decrease the vote tally.